Source/Turbo/Session/Session.swift

```diff
@@ -258,9 +258,13 @@ extension Session: VisitableDelegate {
         }
 
         // Navigating forward - complete navigation early.
-        if visitable === currentVisit.visitable && currentVisit.state == .started {
-            completeNavigationForCurrentVisit()
-            return
+        if visitable === currentVisit.visitable {
```

Comment:
Can we add a code comment explaining this check? I know we can always come back to a PR, but a one-line comment would give immediate context.

p.s. we should revisit using `===` as it's relatively uncommon.

Reply:
Agree.